vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -166,280 +159,262 @@ class OidcSiopWallet(
+        if (!responseType.contains(ID_TOKEN) && !responseType.contains(VP_TOKEN)) {
+            Napier.w("response_type is not supported")
+            return KmmResult.failure(OAuth2Exception(Errors.INVALID_REQUEST))
+        }
+
+        val clientIdSchemeParameters =
+            ClientIdSchemeParametersFactory(request).createClientIdSchemeParameters().getOrElse {
+                return KmmResult.failure(it)
             }
+
+        // TODO Check removed for EUDI interop
+//        if (clientMetadata.subjectSyntaxTypesSupported == null || URN_TYPE_JWK_THUMBPRINT !in clientMetadata.subjectSyntaxTypesSupported) {
+//            return KmmResult.failure<AuthenticationResponseBuilder>(OAuth2Exception(Errors.SUBJECT_SYNTAX_TYPES_NOT_SUPPORTED))
+//                .also { Napier.w("Incompatible subject syntax types algorithms") }
+//        }
+
+        val clientMetadata = runCatching { request.parameters.loadClientMetadata() }.getOrElse {
+            return KmmResult.failure(it)
+        }
+        val clientJsonWebKeySet = clientMetadata.loadJsonWebKeySet()
+
+        val audience = runCatching { request.extractAudience(clientJsonWebKeySet) }.getOrElse {
+            return KmmResult.failure(it)
         }
 
-    private fun RelyingPartyMetadata.requestsEncryption() =
-        authorizationEncryptedResponseAlg != null && authorizationEncryptedResponseEncoding != null
+        val responseModeParameters: ResponseModeParameters =
+            ResponseModeParametersFactory.createResponseModeParameters(
+                request = AuthenticationRequest.createInstance(request),
+            ).getOrElse { return KmmResult.failure(it) }
 
-    private fun authnResponseQuery(
-        request: AuthenticationRequestParametersFrom<*>,
-        response: AuthenticationResponse
-    ): AuthenticationResponseResult.Redirect {
-        if (request.parameters.redirectUrl == null)
-            throw OAuth2Exception(Errors.INVALID_REQUEST)
-        val url = URLBuilder(request.parameters.redirectUrl).apply {
-            response.params.encodeToParameters().forEach {
-                this.parameters.append(it.key, it.value)
+        val presentationDefinition = request.parameters.loadPresentationDefinition()?.also {
+            runCatching {
+                verifyResponseTypeForPresentationDefinition(responseType)
+            }.onFailure { return KmmResult.failure(it) }
+        }
+
+        val presentationPreparationState = presentationDefinition?.runCatching {
+            PresentationPreparationState(
+                presentationDefinition = presentationDefinition,
+                fallbackFormatHolder = clientMetadata.vpFormats
+            ).also {
+                refreshPresentationPreparationState(it)
             }
-        }.buildString()
-        return AuthenticationResponseResult.Redirect(url, response.params)
+        }?.getOrElse { exception ->
+            exception.message?.also { Napier.w { it } }
```

Comment:
Why not simply `Napier.w("presentationPreparationState failed", it)`?

Reply:
fixed in commit b4f3a3714f9c4c39f88da67164cd5aa360e1eaec